function-server/src/main/java/io/dispatchframework/javabaseimage/Server.java

```diff
@@ -118,9 +118,9 @@ public void handleRequest(final HttpServerExchange exchange) {
                         System.out.flush();
                         System.setOut(oldStdout);
 
-                        //TODO add support for stdout - https://github.com/vmware/dispatch/issues/355
                         String[] stderrLogs = baosStderr.toString().length() > 0 ? baosStderr.toString().split("\\r?\\n") : new String[0];
-                        Response response = new Response(new Context(err, stderrLogs), r);
+                        String[] stdoutLogs = baosStdout.toString().length() > 0 ? baosStdout.toString().split("\\r?\\n") : new String[0];
+                        Response response = new Response(new Context(err, new Logs(stderrLogs, stdoutLogs)), r);
```

Comment:
This handler function looks pretty complex. Could you factor out a part of it into a separate method of ExecFunction, that accepts `String message` and returns a `Response`? 

There are 2 positive outcomes of such a refactoring:
- this new function is easily testable so you can also add unit-tests 😉 
- the handler (which is non-unit-testable) would become very small and easy to understand

Reply:
You will be able to test actually printing stuff to err/out and checking where it lands